packages/authentication/src/metadata.ts

```diff
@@ -0,0 +1,15 @@
+// Copyright IBM Corp. 2013,2017. All Rights Reserved.
+// Node module: loopback
+// This file is licensed under the MIT License.
+// License text available at https://opensource.org/licenses/MIT
+
+import * as assert from 'assert';
+import 'reflect-metadata';
+
+export function authenticate(controllerClass: any, methodName: string) {
+  Reflect.defineMetadata('loopback:authenticate', true, controllerClass);
```

Comment:
I think for now at this iteration, I will have to implement at the method level since I need to satisfy the acceptance test given by Ritch. Once that is covered we can extend it further is what I thought.